scanner/utils/hash.go

```diff
@@ -108,24 +109,34 @@ func dirFiles(dir string) ([]string, error) {
 
 // generateHash creates hashes for all files along with filenames and generates a hash for the hashes and filenames.
 func generateHash(files []string, open func(string) (io.ReadCloser, error)) (string, error) {
-	h := sha256.New()
 	files = append([]string(nil), files...)
 	sort.Strings(files)
-	for _, file := range files {
-		if strings.Contains(file, "\n") {
-			return "", errors.New("filenames with newlines are not supported")
+
+	mapper := iter.Mapper[string, string]{
+		MaxGoroutines: len(files) / 2, //nolint:mnd
```

Comment:
This can cause resource problems for large file lists (considering that hashing is CPU bound), I think we should use the default value:

```
// If unset, MaxGoroutines defaults to runtime.GOMAXPROCS(0).
--
MaxGoroutines int
```

Or write a benchmark test with file lists of different sizes.